tests/test_flows.py

```diff
@@ -387,6 +392,49 @@ async def test_run_success_missing_run_name(
         )
         assert result == {"prefect-task": {"cell": "output"}}
 
+    @pytest.mark.respx(assert_all_called=True)
+    async def test_handler_invoked(
+        self, common_mocks, respx_mock, databricks_credentials
+    ):
+        respx_mock.get(
+            "https://dbc-abcdefgh-123d.cloud.databricks.com/api/2.1/jobs/runs/get?run_id=11223344",  # noqa
+            headers={"Authorization": "Bearer testing_token"},
+        ).mock(
+            return_value=Response(
+                200,
+                json={
+                    "state": {
+                        "life_cycle_state": "TERMINATED",
+                        "state_message": "",
+                        "result_state": "SUCCESS",
+                    },
+                    "tasks": [{"run_id": 11223344, "task_key": "prefect-task"}],
+                },
+            )
+        )
+
+        respx_mock.get(
+            "https://dbc-abcdefgh-123d.cloud.databricks.com/api/2.1/jobs/runs/get-output",  # noqa
+            headers={"Authorization": "Bearer testing_token"},
+        ).mock(return_value=Response(200, json={"notebook_output": {"cell": "output"}}))
+        handler = JobSubmissionDummyHandler()
```

Comment:
related to above, if we're handling async then instead of the dummy class i think we could do something like
```python
@pytest.fixture
def global_state():
    return {}

def sync_handler(result, global_state):
    global_state['result'] = result

async def async_handler(result, global_state):
    global_state['result'] = result
    
class TestJobsRunsSubmitAndWaitForCompletion:
    @pytest.mark.parametrize(
        "handler",
        [
            sync_handler,
            async_handler,
            lambda result, state: state.update(result=result)
        ]
    )
    @pytest.mark.respx(assert_all_called=True)
    async def test_handler_invoked(
        self, common_mocks, respx_mock, databricks_credentials
    ):
        ...
        # something like assert len(global_state.get('result') == num_expected_job_runs
```
which also shows that things are being passed through as expected - thoughts?

Reply:
Fixed with [`879e7d2` (#89)](https://github.com/PrefectHQ/prefect-databricks/pull/89/commits/879e7d2a8b7e27680aa61c8b63801d4ea256946b)

Reply:
nice!

Reply:
Apologies there were still minor mistakes